src/mlpack/tests/main_tests/approx_kfn_test.cpp

```diff
@@ -0,0 +1,251 @@
+/**
+ * @file approx_kfn_test.cpp
+ * @author Namrata Mukhija
+ *
+ * Test mlpackMain() of approx_kfn_main.cpp.
+ */
+#include <string>
+
+#define BINDING_TYPE BINDING_TYPE_TEST
+static const std::string testName = "ApproxK-FurthestNeighbors";
+
+#include <mlpack/core.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include "test_helper.hpp"
+#include <mlpack/methods/approx_kfn/approx_kfn_main.cpp>
+
+#include <boost/test/unit_test.hpp>
+#include "../test_tools.hpp"
+
+using namespace mlpack;
+
+struct ApproxKFNTestFixture
+{
+ public:
+  ApproxKFNTestFixture()
+  {
+    // Cache in the options for this program.
+    CLI::RestoreSettings(testName);
+  }
+
+  ~ApproxKFNTestFixture()
+  {
+    // Clear the settings.
+    bindings::tests::CleanMemory();
+    CLI::ClearSettings();
+  }
+};
+
+BOOST_FIXTURE_TEST_SUITE(ApproxKFNMainTest, ApproxKFNTestFixture);
+
+/**
+ * Check that we can't specify both a reference set and an input model.
+ */
+BOOST_AUTO_TEST_CASE(ApproxKFNRefModelTest)
+{
+  arma::mat referenceData;
+  referenceData.randu(2, 80); // 80 points in 2 dimensions.
+
+  // Random input, any k <= reference points.
+  SetInputParam("reference", std::move(referenceData));
+  SetInputParam("k", (int) 10);
+
+  // Input pre-trained model.
+  SetInputParam("input_model", std::move(CLI::GetParam<ApproxKFNModel*>
+    ("output_model")));
```

Comment:
I'm a little bit concerned about this---"output_model" isn't going to be set to anything useful, so I think it may be safer to create an `ApproxKFNModel` in the test then simply pass a pointer to it to `SetInputParam()`.

Also, I think this test means you can check off the first todo in the PR description.

Reply:
Just to clarify on this one, can you change to something like:

```
ApproxKFNModel m;
SetInputParam("input_model", &m);
```